tests/prototypes/workflows.py

```diff
@@ -0,0 +1,122 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+from typing import List, NewType
+
+import scipp as sc
+
+from beamlime.constructors import Factory, ProviderGroup
+
+from .parameters import HistogramBinSize, NumPixels
+
+# Coordinates
+PixelID = NewType("PixelID", sc.Variable)
+EventTimeOffset = NewType("EventTimeOffset", sc.Variable)
+Length = NewType("Length", sc.Variable)
+WaveLength = NewType("WaveLength", sc.Variable)
+
+# Constants
+ConstantA = NewType("ConstantA", sc.Variable)
+ConstantB = NewType("ConstantB", sc.Variable)
+ConstantC = NewType("ConstantC", sc.Variable)
+default_c_a = ConstantA(sc.scalar(0.0001, unit='m'))
+default_c_b = ConstantB(sc.scalar(1, unit='m'))
+default_c_c = ConstantC(sc.scalar(1, unit='1e-6m^2/s'))
+
+# Generated/Calculated
+Events = NewType("Events", List[sc.DataArray])
+MergedData = NewType("MergedData", sc.DataArray)
+PixelIDEdges = NewType("PixelIDEdges", sc.Variable)
+PixelGrouped = NewType("PixelGrouped", sc.DataArray)
+ReducedData = NewType("ReducedData", sc.DataArray)
+Histogrammed = NewType("Histogrammed", sc.DataArray)
+
+
+def workflow_script(
+    da_list: Events, num_pixels: NumPixels, histogram_bin_size: HistogramBinSize
+) -> Histogrammed:
+    c_a = sc.scalar(0.00001, unit='m')
+    c_b = sc.scalar(0.1, unit='m')
+    c_c = sc.scalar(1, unit='1e-3m^2/s')
+
+    merged = sc.concat(da_list, dim='event')
+    pixel_ids = sc.arange(dim='pixel_id', start=0, stop=num_pixels)
+    binned = merged.group(pixel_ids)
+
+    transformed = binned.transform_coords(
+        ['L', 'wavelength'],
+        graph={
+            'L1': lambda pixel_id: (pixel_id * c_a) + c_b,
+            'L2': lambda pixel_id: (pixel_id * c_a) + c_b,
+            'L': lambda L1, L2: L1 + L2,
+            'wavelength': lambda event_time_offset, L: (c_c * event_time_offset / L).to(
+                unit='angstrom'
+            ),
+        },
+    )
+    return transformed.hist(wavelength=histogram_bin_size).sum('L')
+
+
+def merge_data_list(da_list: Events) -> MergedData:
+    return MergedData(sc.concat(da_list, dim='event'))
+
+
+def provide_pixel_id_bin_edges(num_pixels: NumPixels) -> PixelIDEdges:
+    return PixelIDEdges(sc.arange(dim='pixel_id', start=0, stop=num_pixels))
+
+
+def bin_pixel_id(da: MergedData, pixel_bin_coord: PixelIDEdges) -> PixelGrouped:
+    return PixelGrouped(da.group(pixel_bin_coord))
+
+
+def transform_coords(
+    binned: PixelGrouped,
+    c_a: ConstantA = default_c_a,
+    c_b: ConstantB = default_c_b,
+    c_c: ConstantC = default_c_c,
+) -> ReducedData:
+    return ReducedData(
+        binned.transform_coords(
+            ['L', 'wavelength'],
+            graph={
```

Comment:
Is that the same graph as on L49? Can we avoid the duplication?
Maybe this requires `c_a`, `c_b`, and `c_c` to also be part of the graph, but maybe that is a good thing?

Reply:
Yes probably...?
so the idea was to keep `workflow_script` from L36 to L58 to be a through workflow script from the input to the output,
and the rest is granular steps of that workflow, probably run by `sciline` later.
So they are meant to be duplicated.
But the graph might be separated and re-used I agree.